config/object-mapping/src/main/java/io/helidon/config/objectmapping/ObjectConfigMapperProvider.java

```diff
@@ -112,11 +113,21 @@
 
     private static <T> Optional<Function<Config, T>> findStaticStringMethodMapper(Class<T> type,
                                                                                   String methodName) {
-        return findStaticMethod(type, methodName, String.class)
-                .map(handle -> new StringMethodHandleConfigMapper<>(
-                        type,
-                        methodName + "(String) method",
-                        handle));
+
+        Optional<MethodHandle> method = findStaticMethod(type,
+                                                         methodName,
+                                                         String.class);
+
+        if (!method.isPresent()) {
+            method = findStaticMethod(type,
+                                      methodName,
+                                      CharSequence.class);
+        }
+
+        return method.map(handle -> new StringMethodHandleConfigMapper<>(
```

Comment:
Maybe `new CharSequenceMethodHandleConfigMapper<>()`…?

Reply:
It is a method that expects `String` - even though it can now invoke methods with `CharSequence` as a parameter. The generic type is `String`, so I would keep the name